src/banking_stage.rs

```diff
@@ -241,63 +239,163 @@ impl Service for BankingStage {
+        let (banking_stage, _entry_receiver) =
+            BankingStage::new(Arc::new(bank), verified_receiver, None);
+        drop(verified_sender);
+        assert_eq!(banking_stage.join().unwrap(), ());
+    }
+
+    #[test]
+    fn test_banking_stage_tick() {
+        use logger;
+        logger::setup();
+        let bank = Bank::new(&Mint::new(2));
+        let start_hash = bank.last_id();
+        let (verified_sender, verified_receiver) = channel();
+        let (banking_stage, entry_receiver) = BankingStage::new(
+            Arc::new(bank),
+            verified_receiver,
+            Some(Duration::from_millis(1)),
+        );
+        sleep(Duration::from_millis(50));
+        drop(verified_sender);
+
+        let entries: Vec<_> = entry_receiver.iter().map(|x| x).collect();
+        assert!(entries.len() != 0);
+        let mut last_id = start_hash;
+        entries.iter().for_each(|entries| {
+            assert!(entries.verify(&last_id));
+            last_id = entries.last().unwrap().id;
+        });
+        assert_eq!(banking_stage.join().unwrap(), ());
+    }
+
+    #[test]
+    fn test_banking_stage_no_tick() {
+        use logger;
+        logger::setup();
+        let bank = Bank::new(&Mint::new(2));
+        let (verified_sender, verified_receiver) = channel();
+        let (banking_stage, entry_receiver) =
+            BankingStage::new(Arc::new(bank), verified_receiver, None);
+        sleep(Duration::from_millis(1000));
+        drop(verified_sender);
+
+        let entries: Vec<_> = entry_receiver.try_iter().map(|x| x).collect();
+        assert!(entries.len() == 0);
+        assert_eq!(banking_stage.join().unwrap(), ());
+    }
+
+    #[test]
+    fn test_banking_stage() {
+        let mint = Mint::new(2);
+        let bank = Bank::new(&mint);
+        let start_hash = bank.last_id();
+        let (verified_sender, verified_receiver) = channel();
+        let (banking_stage, entry_receiver) =
+            BankingStage::new(Arc::new(bank), verified_receiver, None);
+
+        // good tx
+        let keypair = mint.keypair();
+        let tx = Transaction::new(&keypair, keypair.pubkey(), 1, start_hash);
+
+        // good tx, but no verify
+        let tx_no_ver = Transaction::new(&keypair, keypair.pubkey(), 1, start_hash);
+
+        // bad tx, AccountNotFound
+        let keypair = Keypair::new();
+        let tx_anf = Transaction::new(&keypair, keypair.pubkey(), 1, start_hash);
+
+        // send 'em over
+        let recycler = PacketRecycler::default();
+        let packets = to_packets(&recycler, &[tx, tx_no_ver, tx_anf]);
+
+        // glad they all fit
+        assert_eq!(packets.len(), 1);
+        verified_sender                       // tx, no_ver, anf
+            .send(vec![(packets[0].clone(), vec![1u8, 0u8, 1u8])])
+            .unwrap();
+
+        drop(verified_sender);
+
+        let entries: Vec<_> = entry_receiver.iter().map(|x| x).collect();
```

Comment:
Looks like you should be able to change this to `flat_map()` and remove the loop below it.

Reply:
cool, will do